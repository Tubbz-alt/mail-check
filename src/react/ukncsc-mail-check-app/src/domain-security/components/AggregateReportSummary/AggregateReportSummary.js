import React from 'react';
import startsWith from 'lodash/startsWith';
import values from 'lodash/values';
import { Grid, Header, Loader, Message } from 'semantic-ui-react';
import { items } from 'domain-security/data';
import {
  AggregateReportChart,
  AggregateReportLegendExplanation,
  AggregateReportStatistic,
} from 'domain-security/components';

export default ({ aggregateReportInfo }) => {
  const items1 = values(items);

  const showLoader = aggregateReportInfo.loading;

  const showNoData =
    !aggregateReportInfo.loading &&
    !aggregateReportInfo.error &&
    !aggregateReportInfo.data;

  const showDontHavePermission =
    aggregateReportInfo.error && startsWith(aggregateReportInfo.error, '403');

  const showError =
    aggregateReportInfo.error && !startsWith(aggregateReportInfo.error, '403');

  const showChart =
    !aggregateReportInfo.loading &&
    !aggregateReportInfo.error &&
    aggregateReportInfo.data;

  return (
    <React.Fragment>
      {showLoader && (
        <Header as="h1">
          <Loader active inline style={{ marginRight: 20 }} />
        </Header>
      )}

      {showNoData && (
        <Message info>
          Currently no aggregate report information to show for domain.
        </Message>
      )}

      {showDontHavePermission && (
        <Message info>
          You do have permission to view aggregate reporting for this domain.
        </Message>
      )}

      {showError && (
        <Message error>
          There was a problem retrieving aggregate report information for this
          domain: {aggregateReportInfo.error}
        </Message>
      )}

      {showChart && (
        <React.Fragment>
          <AggregateReportStatistic
            title="Total emails"
            value={aggregateReportInfo.data.totalEmail}
            period="last 7 days"
          />
          <AggregateReportChart data={aggregateReportInfo.data.results} />
          {items1.length > 0 && (
            <Grid stackable>
              <Grid.Row columns={items1.length}>
                {items1.map(_ => (
                  <Grid.Column key={_.title}>
                    <AggregateReportLegendExplanation {..._} />
                  </Grid.Column>
                ))}
              </Grid.Row>
            </Grid>
          )}
        </React.Fragment>
      )}
    </React.Fragment>
  );
};
