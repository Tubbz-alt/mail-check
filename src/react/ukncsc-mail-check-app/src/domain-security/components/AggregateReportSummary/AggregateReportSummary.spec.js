import React from 'react';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';

import AggregateReportSummary from './AggregateReportSummary';

describe('AggregateReportSummary', () => {
  let container;

  describe('when fetching an aggregate report summary', () => {
    beforeEach(() => {
      ({ container } = render(
        <AggregateReportSummary
          aggregateReportInfo={{
            loading: true,
            error: null,
            data: null,
          }}
        />
      ));
    });

    test('it should show loader', () =>
      expect(container.getElementsByClassName('ui loader')).toHaveLength(1));
  });

  describe('when no data returned', () => {
    beforeEach(() => {
      ({ container } = render(
        <AggregateReportSummary
          aggregateReportInfo={{
            loading: false,
            error: null,
            data: null,
          }}
        />
      ));
    });

    test('it should show info message', () =>
      expect(
        container.getElementsByClassName('ui info message')[0]
      ).toHaveTextContent(
        'Currently no aggregate report information to show for domain.'
      ));
  });

  describe('when user doesnt have permission', () => {
    beforeEach(() => {
      ({ container } = render(
        <AggregateReportSummary
          aggregateReportInfo={{
            loading: false,
            error: '403 Forbidden',
            data: null,
          }}
        />
      ));
    });

    test('it should show info message', () =>
      expect(
        container.getElementsByClassName('ui info message')[0]
      ).toHaveTextContent(
        'You do have permission to view aggregate reporting for this domain.'
      ));
  });

  describe('when an error occurs', () => {
    beforeEach(() => {
      ({ container } = render(
        <AggregateReportSummary
          aggregateReportInfo={{
            loading: false,
            error: '404 Not found',
            data: null,
          }}
        />
      ));
    });

    test('it should show error message', () =>
      expect(
        container.getElementsByClassName('ui error message')[0]
      ).toHaveTextContent(
        'There was a problem retrieving aggregate report information for this domain: '
      ));
  });

  describe('when successfully retrieves data', () => {
    beforeEach(() => {
      ({ container } = render(
        <AggregateReportSummary
          aggregateReportInfo={{
            loading: false,
            error: null,
            data: {
              results: {
                '2017-12-10T00:00:00': {
                  fullyTrusted: 4,
                  partiallyTrusted: 6,
                  untrusted: 0,
                  quarantined: 0,
                  rejected: 0,
                },
                '2017-12-11T00:00:00': {
                  fullyTrusted: 5,
                  partiallyTrusted: 8,
                  untrusted: 0,
                  quarantined: 0,
                  rejected: 0,
                },
                '2017-12-12T00:00:00': {
                  fullyTrusted: 6,
                  partiallyTrusted: 7,
                  untrusted: 0,
                  quarantined: 0,
                  rejected: 0,
                },
              },
              totalEmail: 178,
            },
          }}
        />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });
});
