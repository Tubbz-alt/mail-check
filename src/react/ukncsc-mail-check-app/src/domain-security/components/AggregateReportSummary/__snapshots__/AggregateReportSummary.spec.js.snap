// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AggregateReportSummary when successfully retrieves data it should match the snapshot 1`] = `
<div>
  <div
    class="Statistic"
  >
    <h3
      style="margin: 0px;"
      class="ui header StatisticTitle"
    >
      Total emails
    </h3>
    <h1
      style="margin: 0px;"
      class="ui header StatisticValue"
    >
      178.0
    </h1>
    <p
      class="StatisticPeriod"
    >
      last 7 days
    </p>
  </div>
  <div
    class="ui stackable grid"
  >
    <div
      class="two column row"
    >
      <div
        style="padding: 0px;"
        class="thirteen wide column"
      >
        <div>
          <div
            style="position: relative; height: 0px; width: 100%; padding: 0px 0px 57.14285714285714% 0px;"
          >
            <div
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; pointer-events: none;"
              class="VictoryContainer"
            >
              <svg
                width="700"
                height="400"
                role="img"
                aria-labelledby="victory-container-1-title victory-container-1-desc"
                viewBox="0 0 700 400"
                style="pointer-events: all; width: 100%; height: 100%;"
              >
                <g
                  role="presentation"
                >
                  <line
                    x1="50"
                    x2="50"
                    y1="50"
                    y2="350"
                    style="stroke: #252525; stroke-width: 0; fill: transparent; stroke-linecap: round; stroke-linejoin: round;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <g
                    role="presentation"
                  >
                    <line
                      x1="50"
                      x2="650"
                      y1="303.84615384615387"
                      y2="303.84615384615387"
                      style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <line
                      x1="50"
                      x2="49"
                      y1="303.84615384615387"
                      y2="303.84615384615387"
                      style="stroke: transparent; fill: transparent; size: 1px;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <text
                      x="39"
                      dx="0"
                      y="303.84615384615387"
                      dy="4.26"
                    >
                      <tspan
                        x="39"
                        dx="0"
                        text-anchor="end"
                        style="font-size: 12px; font-weight: 500; font-family: 'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                      >
                        2.0
                      </tspan>
                    </text>
                  </g>
                  <g
                    role="presentation"
                  >
                    <line
                      x1="50"
                      x2="650"
                      y1="257.6923076923077"
                      y2="257.6923076923077"
                      style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <line
                      x1="50"
                      x2="49"
                      y1="257.6923076923077"
                      y2="257.6923076923077"
                      style="stroke: transparent; fill: transparent; size: 1px;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <text
                      x="39"
                      dx="0"
                      y="257.6923076923077"
                      dy="4.26"
                    >
                      <tspan
                        x="39"
                        dx="0"
                        text-anchor="end"
                        style="font-size: 12px; font-weight: 500; font-family: 'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                      >
                        4.0
                      </tspan>
                    </text>
                  </g>
                  <g
                    role="presentation"
                  >
                    <line
                      x1="50"
                      x2="650"
                      y1="211.53846153846152"
                      y2="211.53846153846152"
                      style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <line
                      x1="50"
                      x2="49"
                      y1="211.53846153846152"
                      y2="211.53846153846152"
                      style="stroke: transparent; fill: transparent; size: 1px;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <text
                      x="39"
                      dx="0"
                      y="211.53846153846152"
                      dy="4.26"
                    >
                      <tspan
                        x="39"
                        dx="0"
                        text-anchor="end"
                        style="font-size: 12px; font-weight: 500; font-family: 'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                      >
                        6.0
                      </tspan>
                    </text>
                  </g>
                  <g
                    role="presentation"
                  >
                    <line
                      x1="50"
                      x2="650"
                      y1="165.3846153846154"
                      y2="165.3846153846154"
                      style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <line
                      x1="50"
                      x2="49"
                      y1="165.3846153846154"
                      y2="165.3846153846154"
                      style="stroke: transparent; fill: transparent; size: 1px;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <text
                      x="39"
                      dx="0"
                      y="165.3846153846154"
                      dy="4.26"
                    >
                      <tspan
                        x="39"
                        dx="0"
                        text-anchor="end"
                        style="font-size: 12px; font-weight: 500; font-family: 'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                      >
                        8.0
                      </tspan>
                    </text>
                  </g>
                  <g
                    role="presentation"
                  >
                    <line
                      x1="50"
                      x2="650"
                      y1="119.23076923076923"
                      y2="119.23076923076923"
                      style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <line
                      x1="50"
                      x2="49"
                      y1="119.23076923076923"
                      y2="119.23076923076923"
                      style="stroke: transparent; fill: transparent; size: 1px;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <text
                      x="39"
                      dx="0"
                      y="119.23076923076923"
                      dy="4.26"
                    >
                      <tspan
                        x="39"
                        dx="0"
                        text-anchor="end"
                        style="font-size: 12px; font-weight: 500; font-family: 'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                      >
                        10.0
                      </tspan>
                    </text>
                  </g>
                  <g
                    role="presentation"
                  >
                    <line
                      x1="50"
                      x2="650"
                      y1="73.07692307692304"
                      y2="73.07692307692304"
                      style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <line
                      x1="50"
                      x2="49"
                      y1="73.07692307692304"
                      y2="73.07692307692304"
                      style="stroke: transparent; fill: transparent; size: 1px;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <text
                      x="39"
                      dx="0"
                      y="73.07692307692304"
                      dy="4.26"
                    >
                      <tspan
                        x="39"
                        dx="0"
                        text-anchor="end"
                        style="font-size: 12px; font-weight: 500; font-family: 'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                      >
                        12.0
                      </tspan>
                    </text>
                  </g>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    x1="50"
                    x2="650"
                    y1="350"
                    y2="350"
                    style="stroke: #c1c2c4; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <g
                    role="presentation"
                  >
                    <line
                      x1="50"
                      x2="50"
                      y1="350"
                      y2="50"
                      style="stroke: none; stroke-width: 0; fill: none; pointer-events: painted;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <line
                      x1="50"
                      x2="50"
                      y1="350"
                      y2="351"
                      style="stroke: transparent; fill: transparent; size: 1px;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <text
                      x="50"
                      dx="0"
                      y="361"
                      dy="10.26"
                    >
                      <tspan
                        x="50"
                        dx="0"
                        text-anchor="middle"
                        style="font-size: 12px; font-weight: 500; font-family: 'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                      >
                        10th Dec 17
                      </tspan>
                    </text>
                  </g>
                  <g
                    role="presentation"
                  >
                    <line
                      x1="350"
                      x2="350"
                      y1="350"
                      y2="50"
                      style="stroke: none; stroke-width: 0; fill: none; pointer-events: painted;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <line
                      x1="350"
                      x2="350"
                      y1="350"
                      y2="351"
                      style="stroke: transparent; fill: transparent; size: 1px;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <text
                      x="350"
                      dx="0"
                      y="361"
                      dy="10.26"
                    >
                      <tspan
                        x="350"
                        dx="0"
                        text-anchor="middle"
                        style="font-size: 12px; font-weight: 500; font-family: 'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                      >
                        11th Dec 17
                      </tspan>
                    </text>
                  </g>
                  <g
                    role="presentation"
                  >
                    <line
                      x1="650"
                      x2="650"
                      y1="350"
                      y2="50"
                      style="stroke: none; stroke-width: 0; fill: none; pointer-events: painted;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <line
                      x1="650"
                      x2="650"
                      y1="350"
                      y2="351"
                      style="stroke: transparent; fill: transparent; size: 1px;"
                      role="presentation"
                      shape-rendering="auto"
                      vector-effect="non-scaling-stroke"
                    />
                    <text
                      x="650"
                      dx="0"
                      y="361"
                      dy="10.26"
                    >
                      <tspan
                        x="650"
                        dx="0"
                        text-anchor="middle"
                        style="font-size: 12px; font-weight: 500; font-family: 'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                      >
                        12th Dec 17
                      </tspan>
                    </text>
                  </g>
                </g>
                <g>
                  <g
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
                  >
                    <defs>
                      <clippath
                        id="victory-clip-2"
                      >
                        <rect
                          x="50"
                          y="50"
                          width="600"
                          height="300"
                          role="presentation"
                          shape-rendering="auto"
                          vector-effect="non-scaling-stroke"
                        />
                      </clippath>
                    </defs>
                    <g
                      clip-path="url(#victory-clip-2)"
                    >
                      <g>
                        <path
                          d="M50,350L350,350L650,350L650,350L350,350L50,350Z"
                          style="fill: #53ccc2; stroke: none; stroke-width: 2;"
                          role="presentation"
                          shape-rendering="auto"
                        />
                        <path
                          d="M50,350L350,350L650,350"
                          style="fill: none; stroke: #002421; stroke-width: 2;"
                          role="presentation"
                          shape-rendering="auto"
                        />
                      </g>
                    </g>
                  </g>
                  <g
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
                  >
                    <defs>
                      <clippath
                        id="victory-clip-3"
                      >
                        <rect
                          x="50"
                          y="50"
                          width="600"
                          height="300"
                          role="presentation"
                          shape-rendering="auto"
                          vector-effect="non-scaling-stroke"
                        />
                      </clippath>
                    </defs>
                    <g
                      clip-path="url(#victory-clip-3)"
                    >
                      <g>
                        <path
                          d="M50,350L350,350L650,350L650,350L350,350L50,350Z"
                          style="fill: #ff9bf6; stroke: none; stroke-width: 2;"
                          role="presentation"
                          shape-rendering="auto"
                        />
                        <path
                          d="M50,350L350,350L650,350"
                          style="fill: none; stroke: #400038; stroke-width: 2;"
                          role="presentation"
                          shape-rendering="auto"
                        />
                      </g>
                    </g>
                  </g>
                  <g
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
                  >
                    <defs>
                      <clippath
                        id="victory-clip-4"
                      >
                        <rect
                          x="50"
                          y="50"
                          width="600"
                          height="300"
                          role="presentation"
                          shape-rendering="auto"
                          vector-effect="non-scaling-stroke"
                        />
                      </clippath>
                    </defs>
                    <g
                      clip-path="url(#victory-clip-4)"
                    >
                      <g>
                        <path
                          d="M50,350L350,350L650,350L650,350L350,350L50,350Z"
                          style="fill: #ffbf47; stroke: none; stroke-width: 2;"
                          role="presentation"
                          shape-rendering="auto"
                        />
                        <path
                          d="M50,350L350,350L650,350"
                          style="fill: none; stroke: #241800; stroke-width: 2;"
                          role="presentation"
                          shape-rendering="auto"
                        />
                      </g>
                    </g>
                  </g>
                  <g
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
                  >
                    <defs>
                      <clippath
                        id="victory-clip-5"
                      >
                        <rect
                          x="50"
                          y="50"
                          width="600"
                          height="300"
                          role="presentation"
                          shape-rendering="auto"
                          vector-effect="non-scaling-stroke"
                        />
                      </clippath>
                    </defs>
                    <g
                      clip-path="url(#victory-clip-5)"
                    >
                      <g>
                        <path
                          d="M50,211.53846153846152L350,165.3846153846154L650,188.46153846153848L650,350L350,350L50,350Z"
                          style="fill: #93d1ff; stroke: none; stroke-width: 2;"
                          role="presentation"
                          shape-rendering="auto"
                        />
                        <path
                          d="M50,211.53846153846152L350,165.3846153846154L650,188.46153846153848"
                          style="fill: none; stroke: #001f3f; stroke-width: 2;"
                          role="presentation"
                          shape-rendering="auto"
                        />
                      </g>
                    </g>
                  </g>
                  <g
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
                  >
                    <defs>
                      <clippath
                        id="victory-clip-6"
                      >
                        <rect
                          x="50"
                          y="50"
                          width="600"
                          height="300"
                          role="presentation"
                          shape-rendering="auto"
                          vector-effect="non-scaling-stroke"
                        />
                      </clippath>
                    </defs>
                    <g
                      clip-path="url(#victory-clip-6)"
                    >
                      <g>
                        <path
                          d="M50,119.23076923076923L350,50L650,50L650,188.46153846153848L350,165.3846153846154L50,211.53846153846152Z"
                          style="fill: #c4cf81; stroke: none; stroke-width: 2;"
                          role="presentation"
                          shape-rendering="auto"
                        />
                        <path
                          d="M50,119.23076923076923L350,50L650,50"
                          style="fill: none; stroke: #191e00; stroke-width: 2;"
                          role="presentation"
                          shape-rendering="auto"
                        />
                      </g>
                    </g>
                  </g>
                </g>
              </svg>
              <div
                style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
              >
                <svg
                  width="700"
                  height="400"
                  viewBox="0 0 700 400"
                  style="overflow: visible; width: 100%; height: 100%;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="padding: 0px;"
        class="bottom aligned three wide column"
      >
        <ul
          class="ChartLegend"
        >
          <li>
            <h2
              class="ui header ChartLegendHeader"
            >
              -
            </h2>
            <div
              class="ChartLegendMarker"
              style="background: rgb(196, 207, 129); stroke: #c4cf81;"
            />
            Fully trusted
          </li>
          <li>
            <h2
              class="ui header ChartLegendHeader"
            >
              -
            </h2>
            <div
              class="ChartLegendMarker"
              style="background: rgb(147, 209, 255); stroke: #93d1ff;"
            />
            Partially trusted
          </li>
          <li>
            <h2
              class="ui header ChartLegendHeader"
            >
              -
            </h2>
            <div
              class="ChartLegendMarker"
              style="background: rgb(255, 191, 71); stroke: #ffbf47;"
            />
            Untrusted
          </li>
          <li>
            <h2
              class="ui header ChartLegendHeader"
            >
              -
            </h2>
            <div
              class="ChartLegendMarker"
              style="background: rgb(255, 155, 246); stroke: #ff9bf6;"
            />
            Quarantined
          </li>
          <li>
            <h2
              class="ui header ChartLegendHeader"
            >
              -
            </h2>
            <div
              class="ChartLegendMarker"
              style="background: rgb(83, 204, 194); stroke: #53ccc2;"
            />
            Rejected
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    class="ui stackable grid"
  >
    <div
      class="five column row"
    >
      <div
        class="column"
      >
        <h3
          class="ui header"
        >
          Fully trusted
        </h3>
        <p>
          Emails delivered with both successful SPF and DKIM.
        </p>
      </div>
      <div
        class="column"
      >
        <h3
          class="ui header"
        >
          Partially trusted
        </h3>
        <p>
          Emails delivered with either successful SPF or DKIM.
        </p>
      </div>
      <div
        class="column"
      >
        <h3
          class="ui header"
        >
          Untrusted
        </h3>
        <p>
          Emails delivered where SPF and DKIM both failed.
        </p>
      </div>
      <div
        class="column"
      >
        <h3
          class="ui header"
        >
          Quarantined
        </h3>
        <p>
          Emails sent to spam folder because both SPF and DKIM failed.
        </p>
      </div>
      <div
        class="column"
      >
        <h3
          class="ui header"
        >
          Rejected
        </h3>
        <p>
          Emails not delivered to the user because both SPF and DKIM failed.
        </p>
      </div>
    </div>
  </div>
</div>
`;
